src/H5Tref.c

```diff
@@ -753,6 +753,14 @@ H5T__ref_mem_write(H5VL_object_t *src_file, const void *src_buf, size_t src_size
  *
  *-------------------------------------------------------------------------
  */
+/*
+ * It is the correct thing to do to have the reference buffer
+ * be const-qualified here, but the VOL "blob specific" routine
+ * needs a non-const pointer since an operation might modify
+ * the "blob". Disable this warning here (at least temporarily)
+ * for this reason.
+ */
+H5_GCC_CLANG_DIAG_OFF("cast-qual")
```

Comment:
I feel like that could happen in several cases, we should probably fix that VOL callback to handle const and non-const

Reply:
Agreed. Just thought I'd temporarily disable it until we can do that. I'd like to eventually search for all the places we've turned things off with H5_GCC_CLANG_DIAG_OFF in the past and correctly fix things if we can, but that's for a later date.